spyderlib/widgets/tests/test_array_builder.py

```diff
@@ -0,0 +1,204 @@
+
+    def test_array_inline_matrix(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '4 5 6  7 8 9')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+        expected_value = 'np.matrix([[4, 5, 6],\n           [7, 8, 9]])'
+        value = dlg.text()
+        assert expected_value == value
+
+    def test_array_inline_array_invalid(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '1 2  3 4  5 6 7')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return)
+        dlg.update_warning()
+
+        assert not dlg.is_valid()
+
+    def test_array_inline_1d_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '4 5 6')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+        expected_value = 'np.matrix([4, 5, 6])'
+        value = dlg.text()
+        assert expected_value == value
+
+    def test_array_inline_nan_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '4 nan 6 8 9')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+        expected_value = 'np.matrix([4, np.nan, 6, 8, 9])'
+        value = dlg.text()
+        assert expected_value == value
+
+    def test_array_inline_force_float_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True, force_float=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '4 5 6 8 9')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+        expected_value = 'np.matrix([4.0, 5.0, 6.0, 8.0, 9.0])'
+        value = dlg.text()
+        assert expected_value == value
+
+    def test_array_inline_force_float_error_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=True, force_float=True)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClicks(dlg.widget, '4 5 6 a 9')
+        qtbot.keyPress(dlg.widget, Qt.Key_Return, modifier=Qt.ControlModifier)
+
+        expected_value = 'np.matrix([4.0, 5.0, 6.0, a, 9.0])'
+        value = dlg.text()
+        assert expected_value == value
+
+    def test_array_table_array(self, qtbot):  # analysis:ignore
+        dlg = NumpyArrayDialog(inline=False)
+        dlg.show()
+        qtbot.addWidget(dlg)
+
+        qtbot.keyClick(dlg.widget, Qt.Key_1)
+        qtbot.keyClick(dlg.widget, Qt.Key_Tab)
+        qtbot.keyClick(dlg.widget, Qt.Key_2)
+        qtbot.keyClick(dlg.widget, Qt.Key_Tab)
+        qtbot.keyClick(dlg.widget, Qt.Key_Backtab)  # Needed hack
```

Comment:
This is related to the fact that during tests the behavior is not  identical to when launching the actual widget.
In theory when pressing Tab a new cell is created and that becomes the current cell.

In the tests the new cell is created but the current cell is the one after the one it should.
I might need to dig deeper but could not find a solution that did not break the widget